### PR TITLE
Linting and Readme

### PR DESCRIPTION
**Pull request by KristinaMatuleviciute**
Linting done, dependencies and Readme updated

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: node_js
+ + +sudo: false
```
Comment (ckiss) [inline .travis.yml]: please remove the +'es and format the file correctly!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ + +node_js:
+ +   - '5'
+ +   - '4'
+ +
+ + cache:
+ +   directories:
+ +   - node_modules
+ +
+ + after_script:
+ +     npm run coverage
```
Comment (ckiss) [inline .travis.yml]: is _coverage_ defined as script in package.json? if not, you either add it there or remove this _after_script_ from here

File: README.md [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+[![Gitter chat][gitter-badge]][gitter-url]
 
 seneca-hana-store is a [SAP HANA][sapcom] database plugin for the [Seneca][seneca] MVP toolkit. The plugin is using the
 [hdb][nodehdb] driver.
 
-Usage:
+## Install
+
+```sh
+npm install seneca-sqlite-store
```
Comment (ckiss) [inline README.md]: sqlite-store?!

File: README.md [hunk 4]
```diff
@@ -27,7 +41,38 @@ Usage:
       ...
     })
     ...
+    
+## Test
+
+To run tests, simply use npm:
+
+```sh
+npm run test
```
Comment (ckiss) [inline README.md]: does this really work?

File: lib/hana-store.js [hunk 5]
```diff
@@ -1,273 +1,276 @@
-"use strict";
+'use strict'
 
-var _ = require('underscore'),
-    util = require('util'),
-    uuid = require('node-uuid'),
-    hana = require('hdb'),
-    relationalstore = require('./relational-util');
+var _ = require('underscore')
```
Comment (ckiss) [inline lib/hana-store.js]: use const here too instead of var

File: lib/hana-store.js [hunk 6]
```diff
@@ -1,273 +1,276 @@
...
+function quoteval (val) {
+  return '\'' + escapeSQL(val) + '\''
 }
 
 
-function idvalstm(tableName) {
-  var seqName = tableName + '_id';
-  return 'SELECT "' + seqName + '".NEXTVAL from DUMMY';
-}
+/* function never used
```
Comment (ckiss) [inline lib/hana-store.js]: delete when never used

File: lib/hana-store.js [hunk 7]
```diff
@@ -1,273 +1,276 @@
...
-function savestm(ent, tblspec) {
-
-  var table = tblspec.tableName,
-      schema = tblspec.schemaName,
-      entp = relationalstore.makeentp(ent, tblspec),
-      fields = _.keys(entp),
-      cols = [],
-      vals = [];
+function savestm (ent, tblspec) {
+  var table = tblspec.tableName
```
Comment (ckiss) [inline lib/hana-store.js]: use const here too instead of var

File: lib/hana-store.js [hunk 8]
```diff
@@ -1,273 +1,276 @@
...
 
 
-function updatestm(ent, tblspec) {
-  var table = tblspec.tableName,
-      schema = tblspec.schemaName,
-      fields = ent.fields$(),
-      entp = relationalstore.makeentp(ent, tblspec),
-      params = [];
+function updatestm (ent, tblspec) {
+  var table = tblspec.tableName
```
Comment (ckiss) [inline lib/hana-store.js]: same as above

File: lib/hana-store.js [hunk 9]
```diff
@@ -1,273 +1,276 @@
...
+  return 'DELETE FROM ' + tname(schema, table) + wherestm(entp, q, tblspec.columns)
 }
 
 
-function selectstm(qent, q, tblspec) {
-  var table = tblspec.tableName,
-      schema = tblspec.schemaName,
-      entp = relationalstore.makeentp(qent, tblspec);
+function selectstm (qent, q, tblspec) {
+  var table = tblspec.tableName
```
Comment (ckiss) [inline lib/hana-store.js]: same as above

File: lib/hana-store.js [hunk 10]
```diff
@@ -1,273 +1,276 @@
...
 module.exports = function (opts) {
-  var name = 'hana-store',
-      seneca = this,
-      desc,
-      minwait,
-      dbinst = null,
-      specifications = null,
-      tablesmap = {},
-      schema = opts.schema || '';
+  var name = 'hana-store'
```
Comment (ckiss) [inline lib/hana-store.js]: const!

File: lib/hana-store.js [hunk 11]
```diff
@@ -1,273 +1,276 @@
...
-  opts.maxwait = opts.maxwait || MAX_WAIT;
+  opts.minwait = opts.minwait || MIN_WAIT
+  opts.maxwait = opts.maxwait || MAX_WAIT
 
 
-  function gettable(schema, ent, cb) {
-    var canon = ent.canon$({object: true}),
-        table = (canon.base ? canon.base + '_' : '') + canon.name;
+  function gettable (schema, ent, cb) {
+    var canon = ent.canon$({object: true})
```
Comment (ckiss) [inline lib/hana-store.js]: const

File: lib/hana-store.js [hunk 12]
```diff
@@ -276,162 +279,165 @@ module.exports = function (opts) {
...
-        dbinst.disconnect(cb);
+        dbinst.disconnect(cb)
       }
     },
 
 
     save: function (args, cb) {
-      var ent = args.ent,
-          update = !!ent.id;
+      var ent = args.ent
```
Comment (ckiss) [inline lib/hana-store.js]: const

File: lib/hana-store.js [hunk 13]
```diff
@@ -276,162 +279,165 @@ module.exports = function (opts) {
...
-      });
-
+      })
     },
 
-
     load: function (args, cb) {
-      var qent = args.qent,
-          q = args.q;
+      var qent = args.qent
```
Comment (ckiss) [inline lib/hana-store.js]: const

File: lib/hana-store.js [hunk 14]
```diff
@@ -276,162 +279,165 @@ module.exports = function (opts) {
...
+        })
+      })
     },
 
 
     list: function (args, cb) {
-      var qent = args.qent,
-          q = args.q,
-          list = [];
+      var qent = args.qent
```
Comment (ckiss) [inline lib/hana-store.js]: const

File: lib/hana-store.js [hunk 15]
```diff
@@ -276,162 +279,165 @@ module.exports = function (opts) {
...
-      });
+        })
+      })
     },
 
 
     remove: function (args, cb) {
-      var qent = args.qent,
-          q = args.q;
+      var qent = args.qent
```
Comment (ckiss) [inline lib/hana-store.js]: const?

File: lib/hana-store.js [hunk 16]
```diff
@@ -276,162 +279,165 @@ module.exports = function (opts) {
...
 
-  };
+  }
 
 
-  var meta = seneca.store.init(seneca, opts, store);
+  var meta = seneca.store.init(seneca, opts, store)
 
-  desc = meta.desc;
+  // const desc = meta.desc
```
Comment (ckiss) [inline lib/hana-store.js]: why have you commented this? I see it's used bellow!

File: lib/relational-util.js [hunk 17]
```diff
@@ -1,173 +1,192 @@
-/*jslint node: true*/
-"use strict";
+/* jslint node: true*/
```
Comment (ckiss) [inline lib/relational-util.js]: remove this jslint things

File: lib/relational-util.js [hunk 18]
```diff
@@ -1,173 +1,192 @@
...
+'use strict'
 
-var _ = require('underscore'),
-  moment = require('moment');
+var _ = require('underscore')
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: lib/relational-util.js [hunk 19]
```diff
@@ -1,173 +1,192 @@
...
+var Moment = require('moment')
 
-var SENECA_TYPE_COLUMN = 'seneca',
-  BOOLEAN_TYPE = 'b',
-  OBJECT_TYPE = 'o',
-  ARRAY_TYPE = 'a',
-  DATE_TYPE = 'd';
+var SENECA_TYPE_COLUMN = 'seneca'
```
Comment (ckiss) [inline lib/relational-util.js]: same as above

File: lib/relational-util.js [hunk 20]
```diff
@@ -1,173 +1,192 @@
...
+const OBJECT_TYPE = 'o'
+const ARRAY_TYPE = 'a'
+const DATE_TYPE = 'd'
 
-var CHARACTER_TYPES = ['VARCHAR', 'NVARCHAR', 'ALPHANUM', 'SHORTTEXT'],
-  DATETIME_TYPES = ['DATE', 'TIME', 'SECONDDATE', 'TIMESTAMP'],
-  NUMERIC_TYPES = ['TINYINT', 'SMALLINT', 'INTEGER', 'BIGINT', 'SMALLDECIMAL', 'DECIMAL', 'REAL', 'DOUBLE'],
-  BINARY_TYPES = ['VARBINARY'],
-  LOB_TYPES = ['BLOB', 'CLOB', 'NCLOB', 'TEXT'],
+var CHARACTER_TYPES = ['VARCHAR', 'NVARCHAR', 'ALPHANUM', 'SHORTTEXT']
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: lib/relational-util.js [hunk 21]
```diff
@@ -1,173 +1,192 @@
...
+const DATETIME_TYPES = ['DATE', 'TIME', 'SECONDDATE', 'TIMESTAMP']
+const NUMERIC_TYPES = ['TINYINT', 'SMALLINT', 'INTEGER', 'BIGINT', 'SMALLDECIMAL', 'DECIMAL', 'REAL', 'DOUBLE']
+// const BINARY_TYPES = ['VARBINARY']
```
Comment (ckiss) [inline lib/relational-util.js]: if not used, delete, but be 100% sure first

File: lib/relational-util.js [hunk 22]
```diff
@@ -1,173 +1,192 @@
...
-    return false;
+    return false
   }
 
-  var type = dataType.toUpperCase();
+  var type = dataType.toUpperCase()
 
-  return _.indexOf(ALLOWED, type) === -1 ? false : true;
-};
+  return _.indexOf(ALLOWED, type) === -1
```
Comment (ckiss) [inline lib/relational-util.js]: wrong evaluation here.should have been !(_.indexOf(ALLOWED, type) === -1)

File: lib/relational-util.js [hunk 23]
```diff
@@ -239,32 +259,32 @@ module.exports.fixquery = function (entp, q) {
...
 module.exports.makeentp = function (ent, tblspec) {
   if (!ent) {
-    return null;
+    return null
   }
 
-  var entp = {},
-    type = {},
-    fields = ent.fields$();
+  var entp = {}
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: lib/relational-util.js [hunk 24]
```diff
@@ -278,32 +298,30 @@ module.exports.makeentp = function (ent, tblspec) {
...
   if (_.isUndefined(ent) || _.isUndefined(row)) {
-    return null;
+    return null
   }
 
-  var entp = {},
-    fields = _.keys(row),
-    hints = row[SENECA_TYPE_COLUMN],
-    senecatype = hints ? JSON.parse(hints) : {};
+  var entp = {}
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: lib/relational-util.js [hunk 25]
```diff
@@ -278,32 +298,30 @@ module.exports.makeentp = function (ent, tblspec) {
...
+  const fields = _.keys(row)
+  const hints = row[SENECA_TYPE_COLUMN]
+  const senecatype = hints ? JSON.parse(hints) : {}
 
   fields.forEach(function (field) {
     if (SENECA_TYPE_COLUMN !== field) {
-      var cspec = tblspec.columns[field],
-        dataType = (cspec && cspec.dataTypeName) ? cspec.dataTypeName : null;
-      entp[field] = getMapper(dataType).toJS(row[field], senecatype[field]);
+      var cspec = tblspec.columns[field]
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: lib/relational-util.js [hunk 26]
```diff
@@ -278,32 +298,30 @@ module.exports.makeentp = function (ent, tblspec) {
...
 
 
 module.exports.tablename = function (entity) {
-  var canon = entity.canon$({object: true}),
-    name = (canon.base ? canon.base + '_' : '') + canon.name;
-
-  return name;
-};
-
+  var canon = entity.canon$({object: true})
```
Comment (ckiss) [inline lib/relational-util.js]: const

File: package.json [hunk 27]
```diff
@@ -17,18 +17,24 @@
...
-    "async": "0.1.x",
-    "mocha": "1.9.x",
-    "chai": "1.5.x"
+    "async": "1.5.x",
+    "mocha": "2.4.x",
+    "chai": "3.5.x",
+    "eslint-config-seneca": "1.x.x",
+    "eslint-plugin-standard": "1.x.x",
+    "eslint-plugin-hapi": "4.x.x",
+     "lab": "8.2.0"
```
Comment (ckiss) [inline package.json]: fix indentation

File: package.json [hunk 28]
```diff
@@ -17,18 +17,24 @@
...
+  "scripts": {
+    "lint": "lab test -dL"
+  },
   "dependencies": {
-    "seneca": "0.5.11",
-    "hdb": "0.2.0",
-    "underscore": "1.4.x",
+    "seneca": "plugin",
+    "hdb": "0.5.3",
+    "lodash": "~4.x.x",
```
Comment (ckiss) [inline package.json]: so, you replaced underscore with lodash? you need to replace it in every file where it is used then

File: tests/hana.test.js [hunk 29]
```diff
@@ -1,45 +1,44 @@
-"use strict";
+'use strict'
+// var Assert = require('assert')
```
Comment (ckiss) [inline tests/hana.test.js]: delete everything is not used, but be 100% sure before you do

File: tests/hana.test.js [hunk 30]
```diff
@@ -1,45 +1,44 @@
...
-si.__testcount = 0;
-var testcount = 0;
+si.__testcount = 0
+var testcount = 0
 
-var cleanup_cmds = [
-  'DELETE FROM "foo"',
-  'DELETE FROM "moon_bar"'
-];
+// var cleanup_cmds = [
```
Comment (ckiss) [inline tests/hana.test.js]: what is this??

File: tests/hana.test.js [hunk 31]
```diff
@@ -63,28 +62,28 @@ describe('hana', function () {
...
-////        database: 'hana_test',schema: 'hana_test'
-////      }, function (err, data) {
-////        if (err) {
-////          console.error(err);
-////          process.exit(1);
-////        }
-////        console.dir(data);
-////        done()
-////      });
+// //      db.prepare('SELECT * from "egalio"."sys_user"', function (err,stm){
```
Comment (ckiss) [inline tests/hana.test.js]: what is it with all this commented lines?

File: tests/mapper.test.js [hunk 32]
```diff
@@ -1,80 +1,81 @@
...
-    seneca = require('seneca'),
-    async = require('async')
-
-var util = require('../lib/relational-util')
+// var Assert = require('assert')
+// var Seneca = require('seneca')
+// var Async = require('async')
+var Util = require('../lib/relational-util')
 
+var describe
```
Comment (ckiss) [inline tests/mapper.test.js]: you don't need to define describe and it read here: http://webapplog.com/tdd/

File: lib/hana-store.js [hunk 33]
```diff
@@ -1,273 +1,276 @@
...
-var _ = require('underscore'),
-    util = require('util'),
-    uuid = require('node-uuid'),
-    hana = require('hdb'),
-    relationalstore = require('./relational-util');
+var _ = require('underscore')
+const Util = require('util')
+const Uuid = require('node-uuid')
+const Hana = require('hdb')
+const Relationalstore = require('./relational-util')
```
Comment (ckiss) [inline lib/hana-store.js]: use PascalCase here, like RelationalStore

**Comment by ckiss** [comment]
LGTM @mihaidma can you review and merge? thanks

File: lib/hana-store.js [hunk 34]
```diff
@@ -276,162 +271,165 @@ module.exports = function (opts) {
...
+              })
             }
-          });
-        } else {
+          })
+        }
+        else {
           getid(ent, tblspec, function (err, id) {
+            if (err || !id) {
+              return cb(err)
```
Comment (mihaidma) [inline lib/hana-store.js]: Please remove !id from here, it is handle in the next if statement

**Pull request merged**